File: app/src/test/java/com/farmerbb/taskbar/service/UIHostServiceNewControllerTest.java
package com.farmerbb.taskbar.service;

import com.farmerbb.taskbar.ui.DashboardController;
import com.farmerbb.taskbar.ui.StartMenuController;
import com.farmerbb.taskbar.ui.TaskbarController;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricTestRunner;

import static org.junit.Assert.assertTrue;

@RunWith(RobolectricTestRunner.class)
public class UIHostServiceNewControllerTest {
    @Test
    public void testDashboardService() {
        DashboardService dashboardService = Robolectric.setupService(DashboardService.class);
        assertTrue(dashboardService.newController() instanceof DashboardController);
    }

    @Test
    public void testStartMenuService() {
        StartMenuService startMenuService = Robolectric.setupService(StartMenuService.class);
        assertTrue(startMenuService.newController() instanceof StartMenuController);
    }

    @Test
    public void testTaskbarService() {
        TaskbarService taskbarService = Robolectric.setupService(TaskbarService.class);
        assertTrue(taskbarService.newController() instanceof TaskbarController);
    }
}

File: app/src/test/java/com/farmerbb/taskbar/service/UIHostServiceNewControllerTest.kt
package com.farmerbb.taskbar.service

import com.farmerbb.taskbar.ui.DashboardController
import com.farmerbb.taskbar.ui.StartMenuController
import com.farmerbb.taskbar.ui.TaskbarController
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.Robolectric
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class UIHostServiceNewControllerTest {
    @Test
    fun testDashboardService() {
        val dashboardService = Robolectric.setupService(DashboardService::class.java)
        Assert.assertTrue(dashboardService.newController() is DashboardController)
    }

    @Test
    fun testStartMenuService() {
        val startMenuService = Robolectric.setupService(StartMenuService::class.java)
        Assert.assertTrue(startMenuService.newController() is StartMenuController)
    }

    @Test
    fun testTaskbarService() {
        val taskbarService = Robolectric.setupService(TaskbarService::class.java)
        Assert.assertTrue(taskbarService.newController() is TaskbarController)
    }
}
